### DISCUSSION TITLE
Why is this not part of the offical actions and is it specific to GitHub pages?

### DISCUSSION THREAD

**Discussion by KonradHoeffner** [JamesIves/github-pages-deploy-action#1715] [Q&A] [answered]
Building/deploying to a branch in an action is something I do constantly to check out the static HTML of a website on a webserver, however I prefer to use the official actions from <https://github.com/actions/>. Given that this is an incredible basic task, is there a reason why this action is not part of the official GitHub action repositories at <https://github.com/actions/>? Or is this action just for some specific variant of that task and there is a way of doing it using the official actions? The ones I found only upload/deploy to artifacts.

Also, this action includes "pages" in its name but the description seems to state that I can deploy anything to any branch, which part of it is specific to GitHub pages?

**Comment by JamesIves** [comment, marked as answer]
The original intention of this project was to handle GitHub Pages sites and still is for 90% of users, but it can be repurposed for really whatever you want. GitHub Pages can be setup to run from any branch, and renaming the action would be an almost irreversible breaking change so the name has stuck. I use it myself to push files around including my static sites, and also to populate sponsorship information in my README files. 

It's not part of GitHub's Actions org because it's not maintained by GitHub, it's maintained by myself. 👋🏻 GitHub has previously sponsored the project however, but they are not involved in its development.

**Reply by KonradHoeffner** [reply]
Thank you for the swift reply and the clarification!
